src/main/kotlin/dev/arbjerg/lavalink/client/PlayerUpdateBuilder.kt

```diff
@@ -21,30 +18,43 @@ class PlayerUpdateBuilder internal constructor(private val node: LavalinkNode, p
     private var noReplace = false
 
     override fun setTrack(track: Track?): PlayerUpdateBuilder {
-        this.encodedTrack = Omissible.of(track?.encoded)
-        this.trackUserData = track?.userData.toOmissible()
+        this.trackUpdate = PlayerUpdateTrack(
+            encoded = Omissible.of(track?.encoded),
+            userData = track?.userData?.toKotlin().toOmissible()
+        ).toOmissible()
+
         return this
     }
 
     override fun stopTrack(): PlayerUpdateBuilder {
-        this.encodedTrack = Omissible.of(null)
-        this.identifier = null.toOmissible()
-        this.trackUserData = null.toOmissible()
+        this.trackUpdate = PlayerUpdateTrack(
+            encoded = Omissible.of(null),
+        ).toOmissible()
         return this
     }
 
     override fun setEncodedTrack(encodedTrack: String?): PlayerUpdateBuilder {
-        this.encodedTrack = Omissible.of(encodedTrack)
+        this.trackUpdate = PlayerUpdateTrack(
+            encoded = Omissible.of(encodedTrack),
+        ).toOmissible()
         return this
     }
 
+    // TODO: keep this?
     override fun omitEncodedTrack(): PlayerUpdateBuilder {
-        this.encodedTrack = Omissible.omitted()
+        val curTrackUpdate = this.trackUpdate
+
+        if (curTrackUpdate.isPresent()) {
+            this.trackUpdate = curTrackUpdate.value.copy(encoded = Omissible.omitted()).toOmissible()
+        }
+
         return this
     }
 
     override fun setIdentifier(identifier: String?): PlayerUpdateBuilder {
-        this.identifier = identifier.toOmissible()
+        this.trackUpdate = PlayerUpdateTrack(
```

Comment:
should you unset `encoded` in here?

Reply:
Would you say that encoded takes priority over identifier or that its order of execution?

Reply:
Order of execution I'd say

Reply:
Sounds good to me